overlord/configstate/configcore/certs.go

```diff
@@ -0,0 +1,112 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+)
+
+func handleCertConfiguration(tr config.Conf) error {
+	// This handles the "snap revert core" case:
+	// We need to go over each pem cert on disk and check if there is
+	// a matching config entry - if not->delete the cert
+	//
+	// XXX: remove this code once we have a general way to handle
+	//      "snap revert" and config updates
```

Comment:
even before we consider that, there's probably something cheap we can do about detecting if tr is a patch or not. We should do this only if tr is not a patch.  Let's add a TODO

// TODO:  add ways to detect cleanly if tr is a patch, skip the sync code if it is